boards/BMS/Src/Io/Io_CurrentSense.c

```diff
@@ -58,41 +70,40 @@ ExitCode Io_CurrentSense_ConvertToHighResolutionMainCurrent(float adc_voltage, f
     //                                ===
     //                                GND
     //
-    //             HSNBV-D06 Output 2 Voltage - Offset Voltage
-    // Current = ------------------------------------------------
-    //                             Sensitivity
+    //                                                                 1
+    // Current = (HSNBV-D06 Output 1 Voltage - Offset Voltage) x ---------------
+    //                                                             Sensitivity
     //                                              1k + 2.2k
     // HSNBV-D06 Output 2 Voltage = ADC Voltage x --------------
     //                                                 2.2k
     // Offset Voltage = 2.5V
     //
     // Sensitivity = 6.67mV/A
 
-    const float offset             = 2.5f;
-    const float sensitivity        = 6.67e-3f;
-    const float hsnbv_d06_output_2 = adc_voltage * (1e3f + 2.2e3f) / 2.2e3f;
+    // Output from current sensor:
+    const float hsnbv_d06_output_2 = adc_voltage * OUTPUT_2_DIV;
 
-    *high_res_main_current = (hsnbv_d06_output_2 - offset) / sensitivity;
-
-    return EXIT_CODE_OK;
+    // Return the current which corresponds to the output voltage
+    return (hsnbv_d06_output_2 - OUTPUT_2_OFFSET) / OUTPUT_2_SENSITIVITY;
```

Comment:
may be a mistake here with the sensitivity (assuming the #define has been flipped)